### PR TITLE
Make MapR topology compliant to new Clusterdock

### PR DESCRIPTION
**Pull request by kirtiv1**

### CODE REVIEW

File: Dockerfiles/5.2.0/secondary_node/Dockerfile [hunk 1]
```diff
@@ -1,14 +1,13 @@
-FROM cloudera/clusterdock:centos6.6_nodebase
+FROM clusterdock/topology_nodebase:centos6.6
 
-ADD mapr_core.repo /etc/yum.repos.d/
+ADD mapr_core.repo /etc/yum.repos.d/ 
 ADD mapr_ecosystem.repo /etc/yum.repos.d/
 
+RUN yum -y update
```
Comment (dimaspivak) [inline Dockerfiles/5.2.0/secondary_node/Dockerfile]: Why is this needed here and not in the primary_node Dockerfile?

File: Dockerfiles/5.2.0/secondary_node/Dockerfile [hunk 2]
```diff
@@ -1,14 +1,13 @@
-FROM cloudera/clusterdock:centos6.6_nodebase
+FROM clusterdock/topology_nodebase:centos6.6
 
-ADD mapr_core.repo /etc/yum.repos.d/
+ADD mapr_core.repo /etc/yum.repos.d/
```
Comment (dimaspivak) [inline Dockerfiles/5.2.0/secondary_node/Dockerfile]: I think there's a trailing space here.

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,65 @@
+======================================
```
Comment (dimaspivak) [inline README.rst]: Nit: trim the number of equal signs to fit the length of the thing being header-ed.

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+Assuming you've already installed **clusterdock** (if not, go `read the docs`_),
+you use this topology by cloning it to a local folder and then running commands
+with the ``clusterdock`` script:
+
+.. _read the docs: http://clusterdock.readthedocs.io/en/latest/
+
+.. code-block:: console
+
+    $ git clone https://github.com/clusterdock/topology_mapr.git
+    $ clusterdock start topology_mapr --namespace streamsets --node-disks='{node-1:[/dev/xvdb],node-2:[/dev/xvdc]}' --predictable --mapr-version=5.2.0
```
Comment (dimaspivak) [inline README.rst]: I'd replace the equal signs between `--arg` and the argument itself with spaces. Using equals can get hairy in a few edge cases that we actually encounter (e.g. when passing node names to `--secondary-nodes`).

File: start.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```
Comment (dimaspivak) [inline start.py]: This is the wrong header. Use the same one at the top of [start.py from topology_nodebase](https://github.com/clusterdock/topology_nodebase/blob/master/start.py).

File: start.py [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+import logging
+import yaml
+from socket import getfqdn, socket
+from time import sleep, time
+
+from clusterdock.models import Cluster, client, Node
+from clusterdock.utils import wait_for_condition
+
+logger = logging.getLogger('clusterdock.{}'.format(__name__))
```
Comment (dimaspivak) [inline start.py]: Move defining `logger` to be the last thing before `def main`. Also, run `flake8` against this whole repo to find other style issues before submitting the next iteration of this commit.

File: start.py [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+                                                     args.namespace or DEFAULT_NAMESPACE,
+                                                     args.mapr_version)
+    primary_node_image = '{}_{}'.format(image_prefix, 'primary-node')
+    secondary_node_image = '{}_{}'.format(image_prefix, 'secondary-node')
+
+    node_disks = yaml.load(args.node_disks)
+
+    # MapR-FS needs each fileserver node to have a disk allocated for it, so fail fast if the
+    # node disks map is missing any nodes.
+    if set(args.primary_node + args.secondary_nodes) != set(node_disks.keys()):
```
Comment (dimaspivak) [inline start.py]: Can replace `node_disks.keys()` with `node_disks`

File: start.py [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+                       for hostname in args.secondary_nodes]
+
+
+    cluster = Cluster(primary_node, *secondary_nodes)
+    cluster.primary_node = primary_node
+    cluster.start(args.network, pull_images=args.always_pull)
+
+    # Verify that service MapR warden is running and then only proceed.
+    logger.info('Check if Service MapR warden is running ...')
+    start_time = time()
```
Comment (dimaspivak) [inline start.py]: Might be better to use the `wait_for_condition` util function instead of creating your own?

File: start.py [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+    logger.info('Generating new UUIDs ...')
+    cluster.execute('/opt/mapr/server/mruuidgen > /opt/mapr/hostid')
+
+    for node in cluster:
+        configure_command = ('/opt/mapr/server/configure.sh -C {0} -Z {0} -RM {0} -HS {0} '
+                             '-u mapr -g mapr -D {1}'.format(
+                                 primary_node.fqdn, 
+                                 ','.join(node_disks.get(node.hostname))
+                             ))
+        logger.info('Running %s on %s ...', configure_command, node.hostname)
```
Comment (dimaspivak) [inline start.py]: Can remove this; `Node.execute` prints the commands at debug level already.

File: start.py [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+        node.execute("bash -c '{}'".format(configure_command))
+
+    logger.info('Waiting for MapR Control System server to come online ...')
+    def condition(address, port):
+            return socket().connect_ex((address, port)) == 0
```
Comment (dimaspivak) [inline start.py]: Too much indent

File: start.py [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+    def success(time):
+        logger.info('MapR Control System server is online in %s seconds.', time)
```
Comment (dimaspivak) [inline start.py]: => `'MapR Control System server is online after %s seconds.'`

File: start.py [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+    logger.info('Creating /apps/spark directory on %s ...', primary_node.hostname)
+    spark_directory_command = ['sudo -u mapr hadoop fs -mkdir -p /apps/spark',
+                               'sudo -u mapr hadoop fs -chmod 777 /apps/spark']
+    primary_node.execute("bash -c '{}'".format('; '.join(spark_directory_command)))
+
+    logger.info('Creating MapR sample Stream named /sample-stream on %s ...', primary_node.hostname)
+    primary_node.execute('sudo -u mapr maprcli stream create -path /sample-stream '
+                         '-produceperm p -consumeperm p -topicperm p')
+
+    logger.info("MapR Control System server is now accessible at https://%s:%s",
```
Comment (dimaspivak) [inline start.py]: Replace double quotes with single quotes

**Review by dimaspivak — CHANGES_REQUESTED** [review]
Mostly nits. Awesome work, Kirti.

File: start.py (hunk 6 above)
Comment (kirtiv1) [reply start.py]: Done. Ran with flake8.

File: start.py (hunk 7 above)
Comment (kirtiv1) [reply start.py]: Done.

File: start.py (hunk 8 above)
Comment (kirtiv1) [reply start.py]: Done.

File: start.py (hunk 9 above)
Comment (kirtiv1) [reply start.py]: Done.

File: start.py (hunk 10 above)
Comment (kirtiv1) [reply start.py]: Done.

File: start.py (hunk 11 above)
Comment (kirtiv1) [reply start.py]: Done.

File: start.py (hunk 12 above)
Comment (kirtiv1) [reply start.py]: Done.

File: Dockerfiles/5.2.0/secondary_node/Dockerfile (hunk 2 above)
Comment (kirtiv1) [reply Dockerfiles/5.2.0/secondary_node/Dockerfile]: Done.

File: Dockerfiles/5.2.0/secondary_node/Dockerfile (hunk 1 above)
Comment (kirtiv1) [reply Dockerfiles/5.2.0/secondary_node/Dockerfile]: Removed.

File: start.py (hunk 5 above)
Comment (kirtiv1) [reply start.py]: Done.

File: README.rst (hunk 3 above)
Comment (kirtiv1) [reply README.rst]: Done.

File: README.rst (hunk 4 above)
Comment (kirtiv1) [reply README.rst]: Done.

**Review by dimaspivak — APPROVED** [review]
